compiler/rustc_passes/src/dead.rs

```diff
@@ -835,7 +862,9 @@ fn live_symbols_and_ignored_derived_traits(
         ignore_variant_stack: vec![],
         ignored_derived_traits: Default::default(),
     };
-    symbol_visitor.mark_live_symbols();
+    if let ControlFlow::Break(guar) = symbol_visitor.mark_live_symbols() {
+        return Err(guar);
```

Comment:
May also be possible to usr `?` here

Reply:
Unlike mark_live_symbols, live_symbols_and_ignored_derived_traits now returns `Result<(LocalDefIdSet, LocalDefIdMap<FxIndexSet<DefId>>), ErrorGuaranteed>`. So I can't use `?` here. Or do you think it's better to reconsider the return type?

Reply:
Thanks, 
> the best choice considering it would be a very generic and possibly footgunny thing

Let me clarify, does it mean I can keep it as is?
If so, I will squash all my commits then wait for your review/approval!